File: apps/merchant-api/InventoryService/src/InventoryService.Domain/Concretes/InventoryReservation.cs
using InventoryService.Domain.Bases;
using InventoryService.Domain.Enums;

namespace InventoryService.Domain.Concretes;

public class InventoryReservation : BaseEntity
{
    public Guid ClientId { get; set; } = Guid.Empty;
    public ICollection<ProductReservation> Products { get; set; } = new List<ProductReservation>();
    public DateTime SavedDate { get; set; } = DateTime.Now;
    public ReservationStatus ReservationStatus { get; set; } = ReservationStatus.RESERVED;
}

File: apps/merchant-web/src/commons/entities/InventoryReservation.ts
import { UUID } from "crypto";
import VariantStock from "./VariantStock";

export default class InventoryReservation {

  clientId: UUID;
  products: Array<VariantStock>;

  constructor(clientId: UUID, products: Array<VariantStock>) {
    this.clientId = clientId;
    this.products = products;
  }
}